### PR TITLE
Made it possible for real to choose register on io expander for cols and rows

### PR DESCRIPTION
**Pull request by roggan87**
## Description

@ErinCall has made a huge effort to generalize matrix.c for io expander keyboards in #2065. When I tried to put the columns on register B on the io expander I just couldn't get it to work, so I dug into the code. This pretty simple fix solves the issues I faced, and IMHO brings a little bit more clarity to what's going on.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl/config.h [hunk 1]
```diff
@@ -36,7 +36,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define COL_EXPANDED { true, true, true, true, true, true, false, false, false, false, false, false}
 #define MATRIX_ONBOARD_ROW_PINS { F0, F1, F4, F5, F6, F7 }
 #define MATRIX_ONBOARD_COL_PINS { 0, 0, 0, 0, 0, 0, B1, B2, B3, D2, D3, C6 }
-#define EXPANDER_COL_REGISTER 0
+#define EXPANDER_COL_REGISTER GPIOA
```
Comment (ErinCall) [inline keyboards/handwired/dactyl/config.h]: Ah yeah, way better!

File: keyboards/handwired/dactyl/matrix.c [hunk 2]
```diff
@@ -166,25 +162,25 @@ void init_expander(void) {
...
-    | Reg 0 | input, output | output, input |
+    | GPIOA | input, output | output, input |
     +-------+---------------+---------------+
-    | Reg 1 | output, input | input, output |
+    | GPIOB | output, input | input, output |
     +-------+---------------+---------------+
     */
 
-#if (EXPANDER_COLUMN_REGISTER == 0)
+#if (EXPANDER_COL_REGISTER == GPIOA)
```
Comment (ErinCall) [inline keyboards/handwired/dactyl/matrix.c]: Geez, I really wrote half of this feature and committed it without any testing, huh?

Comment (roggan87) [reply keyboards/handwired/dactyl/matrix.c]: I guess so :) Apart from this little typo, you've done a great job and it's a joy working with an expander thanks to your code! To be honest I haven't done any serious testing with the cols on A and rows on B, but in theory it should work.

**Comment by drashna** [comment]
Oh man, this is close to be merged into core....

**Pull request merged**